models/Asset.php

```diff
@@ -548,6 +548,8 @@ public function save(array $parameters = []): static
 
                     break; // transaction was successfully completed, so we cancel the loop here -> no restart required
                 } catch (Exception $e) {
+                    Logger::error('Unable to save Asset: ' . (string) $e);
+
                     try {
                         $this->rollBack();
                     } catch (Exception $er) {
```

Comment:
is the logic below `if ($e instanceof DeadlockException && $retries < ($maxRetries - 1)) {` correct, since we never rebind `$e`